### PR TITLE
[MNG-6566] Plugins that require a certain phase should not fork goals that are already in the execution plan

### PR DESCRIPTION
**Pull request by MartinKanters**
Draft, work needed:
- [X] Implement fix in maven-core, see https://github.com/apache/maven/pull/397
- [ ] ~~Duplicate tests for `@Execute`'s lifecycle and goal parameters~~

### CODE REVIEW

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng6566ExecuteAnnotationShouldNotReExecuteGoalsTest.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            throws Exception
+    {
+        File consumerDir = new File( testDir, "consumer" );
+
+        Verifier verifier = newVerifier( consumerDir.getAbsolutePath() );
+        verifier.setLogFileName( "log-phase-execution.txt" );
+        verifier.executeGoal( "compile" );
+        verifier.resetStreams();
+        verifier.verifyErrorFreeLog();
+        verifyTextNotInLog( verifier, ">>> plugin:1.0-SNAPSHOT:run (default) > compile @ consumer >>>" );
```
Comment (rfscholte) [inline core-it-suite/src/test/java/org/apache/maven/it/MavenITmng6566ExecuteAnnotationShouldNotReExecuteGoalsTest.java:76]: I don't think the forking is the issue, but that the same goal is being executed twice.
What I would do is  a count of all goals being executed and ensure every call is unique.

Comment (MartinKanters) [reply core-it-suite/src/test/java/org/apache/maven/it/MavenITmng6566ExecuteAnnotationShouldNotReExecuteGoalsTest.java:76]: Right, sounds more like what we exactly want to fix in this issue indeed. I will rework it, thanks!

Comment (mthmulders) [reply core-it-suite/src/test/java/org/apache/maven/it/MavenITmng6566ExecuteAnnotationShouldNotReExecuteGoalsTest.java]: We've updated the test to count the invocations of the M-C-P : compile goal.

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng6566ExecuteAnnotationShouldNotReExecuteGoalsTest.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    extends AbstractMavenIntegrationTestCase
+{
+    private static final String RESOURCE_PATH = "/mng-6566-execute-annotation-should-not-re-execute-goals";
+    private static final String PLUGIN_KEY = "org.apache.maven.its.mng6566:plugin:1.0-SNAPSHOT";
+
+    private File testDir;
+
+    public MavenITmng6566ExecuteAnnotationShouldNotReExecuteGoalsTest()
+    {
+        super( "[3.6.3,)" );
```
Comment (rfscholte) [inline core-it-suite/src/test/java/org/apache/maven/it/MavenITmng6566ExecuteAnnotationShouldNotReExecuteGoalsTest.java:40]: Don't think this is the right version

**Review by rfscholte — CHANGES_REQUESTED** [review]
Can't you reuse https://github.com/apache/maven-integration-testing/tree/master/core-it-support/core-it-plugins/maven-it-plugin-core-stubs/maven-compiler-plugin ?

**Review by rfscholte — APPROVED** [review]

**Pull request closed without merging**